### PR TITLE
Many Changes

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: examples/SimpleViewer/ControlPanel.qml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+// SPDX-License-Identifier: LGPL-3.0-or-later
+import QtQuick 2.15
+import QtQuick.Layouts 1.15
+import QtQuick.Controls 2.15
+
+Rectangle {
+    id: root
+
+    color: "white"
```
Comment (jebos) [inline examples/SimpleViewer/ControlPanel.qml]: We should keep properties first in the file, then we set own properties

File: examples/SimpleViewer/ControlPanel.qml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    property alias model: buttonRepeater.model
+    property int lastClicked: -1
+    property bool advandedText: false
+    property alias title: title.text
```
Comment (jebos) [inline examples/SimpleViewer/ControlPanel.qml:15]: Sort them please

File: examples/SimpleViewer/main.qml [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
+// SPDX-FileCopyrightText: 2023 Jonas Kalinka <jonas.kalinka@basyskom.com>
 // SPDX-FileCopyrightText: 2023 basysKom GmbH
-//
-// SPDX-License-Identifier: LGPL-3.0-or-later
 
+// SPDX-License-Identifier: LGPL-3.0-or-later
 import QtQuick 2.15
 import QtQuick.Layouts 1.15
 import QtQuick.Controls 2.15
+import Qt.labs.folderlistmodel 2.15
```
Comment (jebos) [inline examples/SimpleViewer/main.qml]: I am no fan of using the labs module here in an example, can we get rid of it?

File: src/qtquick/datatypes.h [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+    Q_PROPERTY(int id MEMBER id CONSTANT)
+    Q_PROPERTY(QString name MEMBER name CONSTANT)
+
+public:
+    int id;
+    QString name;
+};
+
+struct RenderSettings
```
Comment (jebos) [inline src/qtquick/datatypes.h]: I wonder if this should not be just part of the item, at least we need another name; render settings sound a bit to generic for me.

File: LICENSES/BSD-3-Clause.txt [hunk 5]
```diff
@@ -0,0 +1,9 @@
+Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
```
Comment (jebos) [inline LICENSES/BSD-3-Clause.txt:1]: Why do you add the bsd here?

**Review by jebos — CHANGES_REQUESTED** [review]
The ident changes made it pretty hard to review, next time please keep style changes in a separate pr

File: examples/SimpleViewer/main.qml (hunk 3 above)
Comment (ghost) [reply examples/SimpleViewer/main.qml]: We could hard code the examples instead. I don't think that would be a big deal, especially if there's only ~three.

File: LICENSES/BSD-3-Clause.txt (hunk 5 above)
Comment (ghost) [reply LICENSES/BSD-3-Clause.txt:1]: Only the .clang-format is licensed under BSD

Comment (ghost) [reply LICENSES/BSD-3-Clause.txt:1]: I'd suggest using this one https://github.com/qt-creator/qt-creator/blob/master/.clang-format

File: src/qtquick/datatypes.h (hunk 4 above)
Comment (ghost) [reply src/qtquick/datatypes.h]: I put it into a struct, to have the settings structured and easily handed over to the factory. Especially if there are more coming in the future.
The name is quite generic, but I could not think of a better one.
We could move the Enums, then its not as visible in the QML Item but we could keep the implementation well structured and clear.

File: LICENSES/BSD-3-Clause.txt (hunk 5 above)
Comment (joka42) [reply LICENSES/BSD-3-Clause.txt:1]: I have prepared a branch in which the license is removed and the new formatting is applied.

File: examples/SimpleViewer/main.qml (hunk 3 above)
Comment (jebos) [reply examples/SimpleViewer/main.qml]: yep, do that

File: src/qtquick/datatypes.h (hunk 4 above)
Comment (jebos) [reply src/qtquick/datatypes.h]: At least call it RiveRenderSettings

File: LICENSES/BSD-3-Clause.txt (hunk 5 above)
Comment (jebos) [reply LICENSES/BSD-3-Clause.txt:1]: no I really want to keep the formating as it is. I have a strong mind about formating and how it shall be. 4 ident is okay, but I want to keep the rest as is.

File: src/qtquick/datatypes.h (hunk 4 above)
Comment (joka42) [reply src/qtquick/datatypes.h]: Done

File: examples/SimpleViewer/main.qml (hunk 3 above)
Comment (joka42) [reply examples/SimpleViewer/main.qml]: Done

**Review by jebos — APPROVED** [review]

**Pull request merged**